sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -1646,4 +1651,89 @@ class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAn
       }
     }
   }
+
+  test("ShowTables using v2 catalog") {
+    spark.sql("CREATE TABLE testcat.db.table_name (id bigint, data string) USING foo")
+    spark.sql("CREATE TABLE testcat.n1.n2.db.table_name (id bigint, data string) USING foo")
+
+    runShowTablesSql("SHOW TABLES FROM testcat.db", Seq(Row("db", "table_name", false)))
+
+    runShowTablesSql(
+      "SHOW TABLES FROM testcat.n1.n2.db",
+      Seq(Row("n1.n2.db", "table_name", false)))
+  }
+
+  test("ShowTables using v2 catalog with a pattern") {
+    spark.sql("CREATE TABLE testcat.db.table (id bigint, data string) USING foo")
+    spark.sql("CREATE TABLE testcat.db.table_name_1 (id bigint, data string) USING foo")
+    spark.sql("CREATE TABLE testcat.db.table_name_2 (id bigint, data string) USING foo")
+    spark.sql("CREATE TABLE testcat.db2.table_name_2 (id bigint, data string) USING foo")
+
+    runShowTablesSql(
+      "SHOW TABLES FROM testcat.db",
+      Seq(
+        Row("db", "table", false),
+        Row("db", "table_name_1", false),
+        Row("db", "table_name_2", false)))
+
+    runShowTablesSql(
+      "SHOW TABLES FROM testcat.db LIKE '*name*'",
+      Seq(Row("db", "table_name_1", false), Row("db", "table_name_2", false)))
+
+    runShowTablesSql(
+      "SHOW TABLES FROM testcat.db LIKE '*2'",
+      Seq(Row("db", "table_name_2", false)))
+  }
+
+  test("ShowTables: using v2 catalog, db doesn't exist") {
+    runShowTablesSql("SHOW TABLES FROM testcat.unknown", Seq())
+  }
+
+  test("ShowTables: using v1 catalog, db doesn't exist ") {
+    // 'testcat' below resolves to a database name and falls back to v1 catalog.
+    val exception = intercept[NoSuchDatabaseException] {
```

Comment:
@rdblue here is another difference in v2 catalog. Please compare this with the one right above. AFAIK, in v2 catalog, there is no API to check if a database exists or not. Do we plan to expose one?